File: server/models/delivery.js
/**
 * 配送信息模型 → deliveries 表（订单创建成功后自动生成一条记录）
 *
 * 配送状态常量（status 字段）：
 * 0 待分配 | 1 已分配 | 2 取货中 | 3 配送中 | 4 已送达
 */
const { DataTypes } = require('sequelize')
const sequelize = require('../config/database')

const Delivery = sequelize.define('Delivery', {
  id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
  order_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    unique: true,
    comment: '所属订单 id（一个订单仅一条配送记录）',
  },
  status: {
    type: DataTypes.TINYINT,
    defaultValue: 0,
    comment: '0 待分配 1 已分配 2 取货中 3 配送中 4 已送达',
  },
  courier_name: { type: DataTypes.STRING(50), defaultValue: '', comment: '配送员姓名' },
  courier_phone: { type: DataTypes.STRING(20), defaultValue: '', comment: '配送员电话' },
  estimated_time: { type: DataTypes.DATE, comment: '预计送达时间' },
  delivered_at: { type: DataTypes.DATE, comment: '实际送达时间' },
  timeline: {
    type: DataTypes.TEXT,
    comment: '状态流转时间线 JSON：[{status, text, time}]',
  },
}, {
  tableName: 'deliveries',
})

module.exports = Delivery

File: springboot/src/main/java/com/freshmall/entity/Delivery.java
package com.freshmall.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

/**
 * 配送信息表（一订单一条）
 * 状态：0 待分配 | 1 已分配 | 2 取货中 | 3 配送中 | 4 已送达
 */
@Data
@Entity
@Table(name = "deliveries")
public class Delivery {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "order_id", nullable = false, unique = true)
    private Integer order_id;

    private Integer status = 0;

    @Column(name = "courier_name", length = 50)
    private String courier_name = "";

    @Column(name = "courier_phone", length = 20)
    private String courier_phone = "";

    @Column(name = "estimated_time")
    private LocalDateTime estimated_time;

    @Column(name = "delivered_at")
    private LocalDateTime delivered_at;

    /** 状态流转时间线 JSON：[{status, text, time}] */
    @Column(columnDefinition = "TEXT")
    private String timeline;

    @Column(name = "created_at", updatable = false)
    @CreationTimestamp
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    @UpdateTimestamp
    private LocalDateTime updatedAt;
}
